operator/controllers/operator/eventexporter_controller.go

```diff
@@ -0,0 +1,218 @@
+					Kind:       eventExporter.Kind,
+					Name:       eventExporter.Name,
+					UID:        eventExporter.UID,
+				},
+			},
+			Labels: map[string]string{
+				"app": "eventexporter",
+				"operator.skywalking.apache.org/eventexporter-name": eventExporter.Name,
+				"operator.skywalking.apache.org/application":        "eventexporter",
+				"operator.skywalking.apache.org/component":          "configmap",
+			},
+		},
+		Immutable: &useImmutableConfigMap,
+		Data:      map[string]string{"config.yaml": eventExporter.Spec.Config},
+	}
+
+	if err = r.Client.Create(ctx, &configmap); err != nil {
+		log.Error(err, "failed to create eventexporter's configmap")
+		return true, err
+	}
+
+	return true, nil
+}
+
+func (r *EventExporterReconciler) checkState(ctx context.Context, log logr.Logger, eventExporter *operatorv1alpha1.EventExporter) error {
+	overlay := operatorv1alpha1.EventExporterStatus{}
+	deployment := apps.Deployment{}
+	errCol := new(kubernetes.ErrorCollector)
+
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: eventExporter.Namespace,
+		Name: eventExporter.Name + "-eventexporter"}, &deployment); err != nil && !apierrors.IsNotFound(err) {
+		errCol.Collect(fmt.Errorf("failed to get deployment: %w", err))
+	} else {
+		overlay.Conditions = deployment.Status.Conditions
+		overlay.AvailableReplicas = deployment.Status.AvailableReplicas
+		overlay.ConfigMapName = configMapName(eventExporter)
+	}
+
+	if apiequal.Semantic.DeepDerivative(overlay, eventExporter.Status) {
+		log.Info("Status keeps the same as before")
+		return errCol.Error()
+	}
+
+	if err := r.updateStatus(ctx, eventExporter, overlay, errCol); err != nil {
+		errCol.Collect(fmt.Errorf("failed to update status of EventExporter: %w", err))
+	}
+
+	log.Info("updated Status sub resource")
+
+	return errCol.Error()
+}
+
+func (r *EventExporterReconciler) updateStatus(ctx context.Context, eventExporter *operatorv1alpha1.EventExporter,
+	overlay operatorv1alpha1.EventExporterStatus, errCol *kubernetes.ErrorCollector) error {
+	// avoid resource conflict
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := r.Client.Get(ctx, client.ObjectKey{Name: eventExporter.Name, Namespace: eventExporter.Namespace}, eventExporter); err != nil {
+			errCol.Collect(fmt.Errorf("failed to get EventExporter: %w", err))
+		}
+
+		eventExporter.Status = overlay
+		eventExporter.Kind = "EventExporter"
+		if err := kubernetes.ApplyOverlay(eventExporter, &operatorv1alpha1.EventExporter{Status: overlay}); err != nil {
+			errCol.Collect(fmt.Errorf("failed to apply overlay: %w", err))
+		}
+
+		if err := r.Status().Update(ctx, eventExporter); err != nil {
+			errCol.Collect(fmt.Errorf("failed to update status of EventExporter: %w", err))
+		}
+		return errCol.Error()
+	})
+}
+
+func configMapName(eventExporter *operatorv1alpha1.EventExporter) string {
+	return fmt.Sprintf("%s-eventexporter-cm-%s", eventExporter.Name, MD5Hash(eventExporter.Spec.Config))
+}
```

Comment:
Is it possible that this exceeds the name length limit in k8s?

Reply:
Thx for reply! 

The maximum length of the configMap name is [253 characters](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names), so it is possible to exceed the limit if length of the event exporter name is greater than 221 characters (253 - 32 for config MD5, without other suffix). 

Is it reasonable if I use `eventexporter-cm-{{ md5 }}` for all configMap names where `md5 = Md5Hash(eventExporter.Name + eventExporter.Spec.Config)`? EventExporter name is also hashed to make sure the generated name is unique.

Reply:
In my opinion, it's better to have a fixed length suffix, just like the pods owned by k8s deployments. For the deployment `foo`, one of its pod might look like `foo-84ff48df47-lljb9`.

Reply:
Got it. But I think `-eventexporter-cm-<hash>` is a fixed length suffix since the MD5 hash is always 32 characters regardless of input length. Other hash functions generating fixed length hashes (SHA-1, SHA-256) generate longer outputs compared to MD5. We keep the hashed config as part of the suffix so that we don't need to compare the config content, which cannot be replaced by shorter random strings.

Reply:
OK